Assets/MixedRealityToolkit.Services/DiagnosticsSystem/DiagnosticsSystemVoiceControls.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Diagnostics
+{
+    public class DiagnosticsSystemVoiceControls : MonoBehaviour, IMixedRealitySpeechHandler
+    {
+        private IMixedRealityDiagnosticsSystem diagnosticsSystem = null;
+
+        private IMixedRealityDiagnosticsSystem DiagnosticsSystem
+        {
+            get
+            {
+                if (diagnosticsSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityDiagnosticsSystem>(out diagnosticsSystem);
+                }
+                return diagnosticsSystem;
+            }
+        }
+
+        private IMixedRealityInputSystem inputSystem = null;
+
+        private IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        bool registeredForInput = false;
+
+        private void OnEnable()
+        {
+            if (!registeredForInput)
+            {
+                if (InputSystem != null)
+                {
+                    InputSystem.Register(gameObject);
```

Comment:
I think there's a missing registeredForInput = true statement here (after register is successful)

Reply:
you are correct!